src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -155,6 +155,13 @@ public override bool TryGetValue(int index, string keyName, out object content)
                 return false;
             }
 
+            var issueViz = issues[index];
+
+            if (!(issueViz.Issue is IAnalysisIssue issue))
```

Comment:
It's not obvious that this snapshot is only expecting 'IAnalysisIssue' visualizations.  It might be worth adding an assertion in the constructor to check that all of the vizs are for issues; that might be a bit more obvious.

Reply:
Same applies to the hotspot data source, I guess.